### ISSUE TITLE
Empty 'administrator_login_password' in azurerm_mariadb_server data source

### ISSUE THREAD

**Issue by rmaziarka** [terraform-providers/terraform-provider-azurerm#6521]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

- Terraform: v0.12.24
- AzureRM: v2.6.0

### Affected Resource(s)

- data source `azurerm_mariadb_server`

### Terraform Configuration Files

```hcl
resource "azurerm_mariadb_server" "server" {
  name                = "servername"
  location            = "WestEurope"
  resource_group_name = "resourcegroupname"
  sku_name            = "B_Gen5_1"

  storage_profile {
    storage_mb            = 51200
    backup_retention_days = 7
    geo_redundant_backup  = "Disabled"
  }

  administrator_login          = "admin"
  administrator_login_password = "password"
  version                      = "10.2"
  ssl_enforcement              = "Enabled"
}

data "azurerm_mariadb_server" "datasource" {
  name                = "servername"
  resource_group_name = "resourcegroupname"
}

output "mariadb_server_password" {
  value = data.azurerm_mariadb_server.datasource.administrator_login_password
}
```

### Expected Behavior

In output I should get:
`mariadb_server_password = "password"

### Actual Behavior

I get no information in output.

### Steps to Reproduce

- `terraform apply -auto-approve`

### Additional information

In terraform state file I have null value for password:

```
{
    "schema_version": 0,
    "attributes": {
        "administrator_login": "admin",
        "administrator_login_password": null        
        ..
    }
}
```

**Comment by rmaziarka** [comment]
Azure MariaDB was fixed in the last release ([https://github.com/terraform-providers/terraform-provider-azurerm/issues/6483](link)) but maybe this password was missed.

**Comment by tombuildsstuff** [comment]
hey @rmaziarka 

Thanks for opening this issue.

Taking a look through here the password won't be returned from the Azure API here for security reasons - as such this field probably shouldn't have been added to this data source. As such we should instead look to remove this field from this resource (and double-check the other SQL/MySql/Postgres server data sources to remove this if present, since this field isn't returned from the API and thus'll always be null here.

Thanks!

**Comment by rmaziarka** [comment]
Hey @tombuildsstuff,

Let me disagree with you. For example with data sources:
- [azurerm_kubernetes_cluster](https://www.terraform.io/docs/providers/azurerm/d/kubernetes_cluster.html) you provide all information regarding connection to the cluster - `kube_config` with `username` and `password`
- [cosmosdb_account](https://www.terraform.io/docs/providers/azurerm/d/cosmosdb_account.html) you provide `primary_master_key`
Therefore you provide this information for other services as well. Moreover removing this from the data source would force me to pass the password to the output (I am using a connection to MariaDB cluster from a different state). And it doesn't sound much secure than having it in the data source.

**Comment by tombuildsstuff** [comment]
@rmaziarka different API's behave in different ways - for AKS and CosmosDB (and Storage Accounts etc) they provide a means of obtaining these tokens - unfortunately the Database API's (e.g. MariaDB/MySql/Postgres/SQL Server) don't return this field, so there's nothing for us to output from the API here - which is why this field wants removing

Hope that helps clarify that :)

**Comment by rmaziarka** [comment]
OK, not it is clear :)

**Comment by katbyte** [comment]
closed by #6865
